cidc_api/shared/gcloud_client.py

```diff
@@ -213,15 +251,42 @@ def _get_ttl_expression(days_until_expiry: int = INACTIVE_USER_DAYS) -> str:
     return ttl_expression
 
 
-def _is_matching_conditional_binding(
-    binding: storage.bucket.Policy, member_id: str, subexpression: str
-) -> bool:
-    try:
-        has_user = binding["members"] == {member_id}
-        has_condition = subexpression in binding["condition"]["expression"]
-        return has_user and has_condition
-    except KeyError:
-        return False
+def _get_user_download_binding(
+    policy: storage.bucket.Policy, user_email: str
+) -> Tuple[Optional[int], dict]:
+    """
+    Find a download policy binding for the given `member_id` on `policy`, returning a tuple
+    containing the binding index (possibly `None` if the binding doesn't exist) and the binding 
+    (either existing or newly created).
+    """
+    member_id = f"user:{user_email}"
+
+    # try to find the policy binding on the `policy`
+    user_binding = None
+    user_binding_index = None
+    for i, binding in enumerate(policy.bindings):
+        try:
+            user_is_member = binding["members"] == {member_id}
+            role_is_download = binding["role"] == GOOGLE_DOWNLOAD_ROLE
+            if user_is_member and role_is_download:
+                user_binding = binding
+                user_binding_index = i
```

Comment:
Maybe
```suggestion
                user_binding_index = i
                break
```
? 

Which brings a question of why do we return only one (first, last) binding if user might have multiple?
Shouldn't it raise a flag or try to join multiple together or something?

Reply:
After migrating from the previous implementation of IAM download bindings, a user should have at most one binding that matches this condition. Should the code raise a warning or error if this invariant is violated?

Reply:
yeah, maybe a warning in the logs for us to catch?